hpOneView/image_streamer/resources/artifact_bundles.py

```diff
@@ -0,0 +1,375 @@
+
+    def upload_backup_bundle_from_file(self, file_path, deployment_groups_id_or_uri):
+        """
+        Restore an Artifact Bundle from backup file.
+
+        Args:
+            file_path (str): The File Path to restore the Artifact Bundle.
+            deployment_groups_id_or_uri: ID or URI of the Deployment Groups.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation in
+                OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        if self.DEPLOYMENT_GROUPS_URI not in deployment_groups_id_or_uri:
+            deployment_groups_uri = self.DEPLOYMENT_GROUPS_URI + deployment_groups_id_or_uri
+        else:
+            deployment_groups_uri = deployment_groups_id_or_uri
+
+        upload_file_name = os.path.basename(file_path)
+
+        uri = self.URI + self.BACKUPS_PATH + self.ARCHIVE_PATH + "?deploymentGrpUri=" + deployment_groups_uri
+
+        response, body = self._connection.post_multipart(uri, None, file_path, upload_file_name)
+        if response.status >= 400:
+            raise HPOneViewException(body)
+
+        return body == "SUCCESS"
+
+    def create(self, resource, timeout=-1):
+        """
+        Creates an Artifact Bundle.
+
+        Args:
+            resource (dict): Object to create.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
+                in OneView, just stop waiting for its completion.
+
+        Returns:
+            dict: Created resource.
+        """
+        return self._client.create(resource, timeout=timeout)
+
+    def delete(self, resource, timeout=-1):
+        """
+        Deletes an Artifact Bundle.
+
+        Args:
+            resource(str, dict):
+                Accept either the resource id  or the entire resource.
+            timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
+                in OneView; it just stops waiting for its completion.
+
+        Returns:
+            bool: Indicates if the resource was successfully deleted.
+        """
+        return self._client.delete(resource, timeout=timeout)
+
+    def update(self, id_or_uri, name_to_update, timeout=-1):
+        """
+        Updates only name for the Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            name_to_update: Artifact Bundle name to be updated.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
+                in OneView, just stop waiting for its completion.
+
+        Returns:
+            dict: Updated resource.
+        """
+        if self.URI not in id_or_uri:
+            id_or_uri = self.URI + "/" + id_or_uri
+
+        data = {
```

Comment:
We usually don't build the dictionaries for update. It is the standard of the SDK update methods.